File: leet_2418_sort_people.go
func sortPeople(names []string, heights []int) []string {
    N := len(names)
    nhs := make([]NH, N)
    // i := 0
    // for i < N {
    for i, name := range names {
        nhs[i] = NH{name, heights[i]}
        i++
    }
    sort.Slice(nhs, func(l, r int) bool {
        return nhs[l].h > nhs[r].h
    })
    res := make([]string, N)
    for i, nh := range nhs {
        res[i] = nh.n
    }
    return res//[]string{}
}

type NH struct {
    n string
    h int
}

File: leet_2418_sort_people.rs
impl Solution {
    pub fn sort_people(names: Vec<String>, heights: Vec<i32>) -> Vec<String> {

        use std::cmp::Ordering;
        struct NH { n:String, h:i32 }
        let mut nhs : Vec<NH> = names
            .into_iter()//iter()
            .zip(heights.into_iter()/*iter()*/)
            .map( | (n, h) | NH {n, h} )
            .collect();
        nhs.sort_by( | l, r | r.h.cmp(&l.h) );
        nhs.into_iter()/*iter()*/.map( | nh | nh.n ).collect()//;
    }
}
